### DISCUSSION TITLE
Firefox finds information in an image that ImageMagick doesn't

### DISCUSSION THREAD

**Discussion by ArthurPeabody** [ImageMagick/ImageMagick#7787] [Help]
### ImageMagick version

7.1.1-41

### Operating system, version and so on

Linux, Slackware (current), Kernel 6.12

### Description

 A vendor e-mailed me a return label (in pdf). When I cropped the mailing label portion, rotated it and resized it with imagemagick, it became unreadably vague, as though it made space between the pixels that were in the original. So I resized and cropped it with Firefox, which was easily readable - which means the information was there. I rotated it with imagemagick successfully. It seems that Firefox found information that imagemagick didn't.

**Comment by fmw42** [comment]
Post the PDF and your command line.  Possibly, it is of low resolution and needs a higher resolution (-density value) when reading in the PDF.

**Reply by ArthurPeabody** [reply]
I used display, not a command line.  It's difficult to figure out where to set the borders to crop an image otherwise.  The return label has personal information.  How do I figure out the original density?  I ran identify, got a resolution of 72x72 - is that the same as density?

**Reply by fmw42** [reply]
Post the PDF.

**Reply by ArthurPeabody** [reply]
No.  It has my address.

**Reply by fmw42** [reply]
You said you cropped, rotated and resized.  Please show your command line.

**Comment by snibgo** [comment]
I suggest converting the PDF to a simple raster image format, like this:
```
magick -density 300 in.pdf out.png
```
Then you can edit out.png with an interactive image editor like Gimp to crop, rotate and resize as you wish.

You may need to adjust the density number to increase resolution, eg 600 or 1200.

EDIT: Corrected position of "-density 300".

**Reply by fmw42** [reply]
I believe that should be

`magick -density 300 in.pdf out.png
`

or

`magick -density 300 in.pdf -crop ... -rotate ... -resize ... out.png`

**Reply by snibgo** [reply]
Oops, yes, thanks for the correction.

**Reply by ArthurPeabody** [reply]
magick input.pdf -crop 405x542x300x300 -rotate 270 -resize 810x1084 -density 300 output.pdf
Any density argument makes the output a blank file. The key seems to be PDF-ness. Firefox can resize PDFs and make them more readable. It can't do that for other image files. I can't extract any text with poppler's pdftotext or mark it in Firefox, as I can with PDFs composed from text sources. Firefox expands it, then improves the image as I watch. It's all text, line art, a QR code and a UPC code, so it's model-able as a non-image.  I can't read the original; ImageMagick and gimp expand the unreadability; Firefox makes it readable.

Converting to an image file and manipulating with gimp doesn't help.  Firefox 'knows' something about PDFs that it doesn't about images; it seems ImageMagick and gimp aim at image files.

**Reply by fmw42** [reply]
Try

`magick -density 288 input.pdf -resize 25% -crop 405x542x300x300 -rotate 270 -resize 810x1084 -density 300 output.pdf`

Adjust your crop and resize if needed.  Does that help?

**Reply by ArthurPeabody** [reply]
No.  Any density statement results in a blank output.  Omitting the density statements makes an even fuzzier output.

**Comment by snibgo** [comment]
I understand if you can't post the PDF. Instead, please post the text output from:
```
magick -verbose -density 288 input.pdf info:
```
... and again, without the "-density 288".

**Reply by ArthurPeabody** [reply]
'gs' -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r288x288' -dPrinted=false  '-sOutputFile=/tmp/magick-bw0waSqtP2kYij676PnwKhOSzyaYKmY1%d' '-f/tmp/magick-57NDvDdiFyBR7y-atg1qgi_YjLWXdYmE' '-f/tmp/magick-UHS6sH9Uel9kzf2nOpY2GcwbWmOp6wle'
/tmp/magick-bw0waSqtP2kYij676PnwKhOSzyaYKmY11 PNG 2380x3368 2380x3368+0+0 8-bit TrueColorAlpha sRGB 439339B 0.230u 0:00.179
ReturnLabel_AltecLansingSpeakers_Page1-20241116.pdf PDF 2380x3368 2380x3368+0+0 16-bit TrueColorAlpha sRGB 439339B 0.000u 0:00.000
ReturnLabel_AltecLansingSpeakers_Page1-20241116.pdf=>info PDF 2380x3368 2380x3368+0+0 16-bit Grayscale Gray 467012B 0.170u 0:00.117


'gs' -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -dPrinted=false  '-sOutputFile=/tmp/magick-SjI2gdNHDKW4fIhGdnx-gFJ2MOVsAhPg%d' '-f/tmp/magick-oAI2IT5HyFUzPhp5ZAIewiwjFyBELZ0v' '-f/tmp/magick-SbbVhEAUWTjbfq_E6hj7lzZ9HyFoc5Vu'
/tmp/magick-SjI2gdNHDKW4fIhGdnx-gFJ2MOVsAhPg1 PNG 595x842 595x842+0+0 8-bit TrueColorAlpha sRGB 37767B 0.010u 0:00.011
ReturnLabel_AltecLansingSpeakers_Page1-20241116.pdf PDF 595x842 595x842+0+0 16-bit TrueColorAlpha sRGB 37767B 0.000u 0:00.000
ReturnLabel_AltecLansingSpeakers_Page1-20241116.pdf=>info PDF 595x842 595x842+0+0 16-bit Grayscale Gray 33965B 0.020u 0:00.007

**Comment by snibgo** [comment]
I expect loads more text, starting something  like this:
```
Image:
  Filename: output.pdf
  Format: PDF (Portable Document Format)
  Mime type: application/pdf
  Class: DirectClass
  Geometry: 2448x3168+0+0
  Resolution: 288x288
  Print size: 8.5x11
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColorAlpha
  Endianness: Undefined
  Depth: 16/8-bit
  Channels: 4.0
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
    Alpha: 1-bit
  Channel statistics:
    Pixels: 7755264
    Red:
      min: 0  (0)
      max: 65535 (1)
      mean: 40725 (0.621423)
      median: 65535 (1)
      standard deviation: 30394.2 (0.463786)
      kurtosis: -1.67094
      skewness: -0.505836
      entropy: 0.311436
    Green:
```
That text may help to explain why "Any density statement results in a blank output".
